sdk/src/main/java/io/radar/sdk/RadarBeaconUtils.kt

```diff
@@ -15,7 +15,7 @@ internal object RadarBeaconUtils {
     private const val MANUFACTURER_ID = 76
 
     fun getScanFilter(beacon: RadarBeacon): ScanFilter? {
-        val uuid = UUID.fromString(beacon.uuid)
+        val uuid = UUID.fromString(beacon.uuid.toLowerCase())
```

Comment:
Does it work without lowercasing the UUID?  Would be nice to leave the value unchanged if Android beacon scanning is resilient to mismatched cases. Just thinking about whether we'll start opening ourselves up to bugs if the beacon UUID case doesn't match across 1) the hardware, 2) what's stored in the server, 3) what the client searches for

Reply:
We should be able to remove this line. Constructing the uuid handles lowercase and uppercase.

This is the important line: https://github.com/radarlabs/radar-sdk-android/pull/119/files#diff-c1686d96eea909dad7c45aa1494dd7f80a93a446ada571c2b9249fa6968dfbbdR67 When we extract the uuid from the scan result and convert it to a string, Android gives us an all-lowercase string, which may not match the uuid string on our server. This line makes the comparison case insensitive.

Reply:
UUIDs are just 128-bit values represented as hex strings. Wen it comes to the value of the UUID -- input or beacon ranging -- the casing doesn't matter since it's just hex. It's only an issue when we're doing string comparisons.

Reply:
@tjulien Another option is to compare UUIDs directly instead of doing string comparisons. Let me test that.